tcms/testruns/tests/test_permissions.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name, too-many-ancestors
+
+from http import HTTPStatus
+
+from django.urls import reverse
+
+from tcms.tests import PermissionsTestCase
+from tcms.tests import remove_perm_from_user
+from tcms.tests import user_should_have_perm
+from tcms.tests import factories
+
+
+class TestEditRun(PermissionsTestCase):
+    """Test edit view method"""
+
+    permission_label = 'testruns.change_testrun'
+    http_method_names = ['get', 'post']
+    url = 'url'
+    post_data = {
+        'summary': '',
+        'build': '',
+        'manager': '',
+        'default_tester': '',
+        'notes': '',
+    }
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.product = factories.ProductFactory(name='KiwiTCMS')
```

Comment:
Some of these are probably only needed to create other objects and assign values to `post_data`. Not all of them need to be class attributes. They can be ordinary variables in this method.